src/Kiota.Builder/Writers/Php/CodePropertyWriter.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Diagnostics;
+using System.Linq;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder.Writers.Php
+{
+    public class CodePropertyWriter: BaseElementWriter<CodeProperty, PhpConventionService>
+    {
+        public CodePropertyWriter(PhpConventionService conventionService) : base(conventionService) { }
+
+        public override void WriteCodeElement(CodeProperty codeElement, LanguageWriter writer)
+        {
+            
+            var returnType = conventions.GetTypeString(codeElement.Type, codeElement);
+            var currentPathProperty = codeElement.Parent.GetChildElements(true)
+                .OfType<CodeProperty>()
+                .FirstOrDefault(x => x.IsOfKind(CodePropertyKind.PathParameters));
+            var propertyName = codeElement.Name.ToFirstCharacterLowerCase();
+            var propertyAccess = conventions.GetAccessModifier(codeElement.Access);
+            switch (codeElement.PropertyKind)
+            {
+                case CodePropertyKind.RequestBuilder:
+                    conventions.WriteShortDescription(codeElement.Description, writer);
+                    writer.WriteLine($"{propertyAccess} function {propertyName}(): {returnType} {{");
+                    writer.IncreaseIndent();
+                    conventions.AddRequestBuilderBody(currentPathProperty != null, returnType, writer);
+                    writer.DecreaseIndent();
+                    writer.WriteLine("}");
+                    break;
+                case CodePropertyKind.RequestAdapter:
+                    WritePropertyDocComment(codeElement, writer);
+                    writer.WriteLine($"{propertyAccess} RequestAdapter ${propertyName};");
+                    break;
+                case CodePropertyKind.AdditionalData or CodePropertyKind.PathParameters:
+                    WritePropertyDocComment(codeElement, writer);
+                    writer.WriteLine($"{propertyAccess} array ${propertyName};");
```

Comment:
should rely on the type of the dom element, which can be updated by a refiner prior.